app/pages/patientdata/patientdata.js

```diff
@@ -504,13 +485,20 @@ export let PatientData = React.createClass({
 
     var refresh = this.props.onRefresh;
     if (refresh) {
-      this.setState({title: this.DEFAULT_TITLE});
-      refresh();
+      this.props.clearPatientData(this.props.routeParams.id);
+      this.setState({ 
+        title: this.DEFAULT_TITLE, 
+        processingData: true,
+        processedPatientData: null 
```

Comment:
Should set the other things we are deriving after processing to `null` too - `bgPrefs` &c.